samples/experimental/ExperimentalProvider/ExperimentalProviderApp/MainWindow.xaml.cs

```diff
@@ -118,5 +118,41 @@ public async Task<string> AddUriAsync(UriActionEntity uriAction)
 
             return await tcs.Task;
         }
+
+        public async Task<string> AddCustomTextAsync(CustomTextActionEntity customTextAction)
+        {
+            TaskCompletionSource<string> tcs = new();
+            DispatcherQueue.TryEnqueue(() =>
+            {
+                string messageContent = "";
+
+                if(customTextAction.KeyPhrase != null)
+                {
+                    messageContent += customTextAction.KeyPhrase;
+                }
+                Messages.Add(messageContent);
+                tcs.SetResult(string.Format(resourceLoader.GetString("CustomTextMessageResult"), Messages.Count.ToString()));
+            });
+
+            return await tcs.Task;
+        }
+
+        public async Task<string> AddStreamingTextAsync(StreamingTextActionEntity streamingTextAction)
+        {
+            TaskCompletionSource<string> tcs = new();
+            DispatcherQueue.TryEnqueue(() =>
+            {
+                string messageContent = "";
```

Comment:
How does this display? Looks like it would be something like

This
This is
This is a 
This is a streaming
This is a streaming text